libs/application/template-api-modules/src/lib/modules/shared/data-providers/user-profile/user-profile.service.ts

```diff
@@ -1,45 +1,135 @@
     private readonly userProfileApi: UserProfileApi,
-  ) {}
+    private readonly islyklarService: IslykillService,
+  ) { }
 
-  userProfileApiWithAuth(auth: Auth) {
+  userProfileApiWithAuth(auth: Auth): UserProfileApi {
     return this.userProfileApi.withMiddleware(new AuthMiddleware(auth))
   }
 
-  async getUserProfile({ auth }: TemplateApiModuleActionProps) {
-    try {
-      //TODO finish this
-      const profile = await this.userProfileApiWithAuth(
-        auth,
-      ).userProfileControllerFindOneByNationalId({
-        nationalId: auth.nationalId,
+  // eslint-disable-next-line
+  async getIslykillProfile(auth: Auth) {
+    if (auth.nationalId === undefined) {
+      return null
+    }
+
+    this.islyklarService.getIslykillSettings(auth.nationalId)
+      .then(islyklarData => {
+        return {
+          nationalId: auth.nationalId,
+          emailVerified: false,
+          mobilePhoneNumberVerified: false,
+          documentNotifications: false,
+          emailStatus: DataStatus.NOT_VERIFIED,
+          mobileStatus: DataStatus.NOT_VERIFIED,
+
+          // Islyklar data:
+          mobilePhoneNumber: islyklarData?.mobile,
+          email: islyklarData?.email,
+          canNudge: islyklarData?.canNudge,
+          bankInfo: islyklarData?.bankInfo,
+        }
+      }).catch(error => {
+        logger.error(JSON.stringify(error))
+        return null
       })
+  }
+
+  async getUserProfileStatus(auth: Auth) {
+    /**
+     * this.getUserProfile can be a bit slower with the addition of islyklar data call.
+     * getUserProfileStatus can be used for a check if the userprofile exists, or if the userdata is old
+     * Old userdata can mean a user will be prompted to verify their info in the UI.
+     */
+    if (auth.nationalId === undefined) {
+      return null
+    }
+
+    this.userProfileApiWithAuth(auth)
+      .userProfileControllerFindOneByNationalId({ nationalId: auth.nationalId })
+      .then(profile => {
+        /**
+         * If user has empty email or tel data
+         * Then the user will be prompted every 6 months (MAX_OUT_OF_DATE_MONTHS)
+         * to verify if they want to keep their info empty
+         */
+        const emptyMail = profile?.emailStatus === 'EMPTY'
+        const emptyMobile = profile?.mobileStatus === 'EMPTY'
+        const modifiedProfileDate = profile?.modified
+        const dateNow = new Date()
+        const dateModified = new Date(modifiedProfileDate)
+        const diffInMonths = differenceInMonths(dateNow, dateModified)
+        const diffOutOfDate = diffInMonths >= MAX_OUT_OF_DATE_MONTHS
+        const outOfDateEmailMobile = (emptyMail || emptyMobile) && diffOutOfDate
 
-      return {
-        email: profile.email,
-        emailVerified: profile.emailVerified,
-        mobilePhoneNumber: profile.mobilePhoneNumber,
-        mobilePhoneNumberVerified: profile.mobilePhoneNumberVerified,
-      }
-    } catch (e) {
-      if (isRunningOnEnvironment('local')) {
```

Comment:
We dont want everyone working on the application system having to set up and running the userprofile service and or islykill.